lib/correlations/client.go

```diff
@@ -59,6 +57,7 @@ type request struct {
 // Client is a client for making dimensional correlations
 type Client struct {
 	sync.RWMutex
+	log           *zap.Logger
```

Comment:
Not a big fan of pulling in another logging library just because collector happens to use it.  Maybe try and make the interface return errors as appropriate and let the users of the library print errors with whatever log library they want.  Not sure if that is possible given the data flows.

Reply:
Yeah not really possible to return errors as there's a lot of informational messages as well. I think the only option is having a generic logging interface similar to slf4j in the Java world. I'll see if I can find one that's already out there we can use.

Reply:
The logging library wrapper I found wasn't able to work with how we pass zap loggers around in OT so I just added a simple wrapper instead.